doc/contribute/index.rst

```diff
@@ -69,6 +69,50 @@ this contributing and review process for imported components.
 
       ../LICENSING.rst
 
+Binary blobs
+============
+
+Any contributions to the Zephyr project must be in source code form.
+Contributions which depend on proprietary, binary components, such as
+proprietary libraries, are not allowed. Such components are generally referred
+to as *binary blobs*.
+
```

Comment:
The status of components, contributed as open source code that are dependent, or made to work with, proprietary binary components is not clear.
For instance, what is the status of fully open source driver that would be made to communicate with a proprietary co-processor firmware ?

Reply:
How is that different from an OSS driver communicating with a proprietary IP block? Or a flash device, which often contains proprietary wear leveling firmware?

Reply:
This could be the same, point is I have not been able to identify in which category this falls vs the binary blob discussion and proposed allowed typed. Cf my mail to TSC.
So if that's the same, what is the status ?

Reply:
> How is that different from an OSS driver communicating with a proprietary IP block? 

The big difference is: "Does it work out of the box?"

In longer words: "Does it work without having to _download_ some binaries and/or agree to some additional license and legal terms?". I think this paragraph must make that clearer.

> Or a flash device, which often contains proprietary wear leveling firmware?

While they all have some proprietary firmware[*] an SD card, eMMC or memory stick never require a legal agreement. You make a very good point: the difference between hardware and firmware has been completely blurred. But there is still one big difference and clear line: there is still a lot of "hardware" that you can use without a lawyer (neither a phone nor a car unfortunately)

[*] nice article https://www.bunniestudios.com/blog/?page_id=3592

Reply:
@erwango I responded on the TSC list and added a parenthetical comment here trying to clarify the situation. Hope this covers your doubt.

Reply:
> I responded on the TSC list

https://lists.zephyrproject.org/g/tsc/message/1226